atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java

```diff
@@ -515,4 +497,20 @@ static boolean isRetriableException(Throwable t) {
                 || isRetriableException(t.getCause()));
     }
 
+    final FunctionCheckedException<Cassandra.Client, Void, Exception> validatePartitioner = new FunctionCheckedException<Cassandra.Client, Void, Exception>() {
+        @Override
+        public Void apply(Cassandra.Client client) throws Exception {
+            CassandraVerifier.validatePartitioner(client, config);
+            return null;
+        }
+    };
+
+    final FunctionCheckedException<Cassandra.Client, Void, Exception> createInternalMetadataTable = new FunctionCheckedException<Cassandra.Client, Void, Exception>() {
+        @Override
+        public Void apply(Cassandra.Client client) throws Exception {
+            createTableInternal(client, CassandraConstants.METADATA_TABLE);
+            return null;
+        }
+    };
```

Comment:
not trivially, they need non-static Cassandra config and it's not easy to pass in both to a FunctionCheckedException